front/components/vaults/VaultWebsiteModal.tsx

```diff
@@ -133,10 +142,10 @@ export default function VaultCreateWebsiteModal({
   };
 
   useEffect(() => {
-    if (isUrlValid(dataSourceUrl)) {
+    if (isUrlValid(dataSourceUrl) && !dataSourceName) {
```

Comment:
We should not check for `!dataSourceName` as it will stop updating the data source name as soon as the url will be considered valid. See PR that fixed this a few months ago [here](https://github.com/dust-tt/dust/pull/6164).

Reply:
I replaced by dataSourceView then. We don't want datasourceName state to be updated if the website exists.